packages/near-fast-auth-signer/src/components/AuthCallback/AuthCallback.tsx

```diff
@@ -1,290 +1,193 @@
-          lakPublicKey: public_key_lak,
-          gateway,
-          accountId:    accountIds[0],
-        });
-
-        setStatusMessage('Account recovered successfully!');
-
-        setStatusMessage('Redirecting to app...');
-
-        const parsedUrl = new URL(
-          success_url && isUrlNotJavascriptProtocol(success_url)
-            ? success_url
-            : window.location.origin + (basePath ? `/${basePath}` : '')
-        );
-        parsedUrl.searchParams.set('account_id', accountIds[0]);
-        parsedUrl.searchParams.set('public_key', public_key_lak);
-        parsedUrl.searchParams.set('all_keys', (publicKeyFak ? [public_key_lak, publicKeyFak, recoveryPK] : [public_key_lak, recoveryPK]).join(','));
+const AccountInfo = styled.div<{ textCentered?: boolean }>`
+  width: 375px;
+  display: flex;
+  padding: 8px 12px;
+  align-items: center;
+  justify-content: center;
+  gap: 6px;
+  border-radius: 4px;
+  border: 1px solid #FBD38D;
+  background: #FEEBC8;
+  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.06);
+  @media (min-width: 768px) {
+    max-width: 380px;
+  }
+  margin-bottom: 12px;
+  color: #2D3748;
+  p{
+    margin: 0;
+  }
+`;
 
-        if (inIframe()) {
-          window.parent.postMessage({
-            type:   'method',
-            method: 'query',
-            id:     1234,
-            params: {
-              request_type: 'complete_authentication',
-              publicKey:    public_key_lak,
-              allKeys:      (publicKeyFak ? [public_key_lak, publicKeyFak, recoveryPK] : [public_key_lak, recoveryPK]).join(','),
-              accountId:    accountIds[0]
-            }
-          }, '*');
-        } else {
-          window.location.replace(parsedUrl.href);
-        }
-      }
-    });
-};
+const StatusMessage = styled.p`
+`;
 
 function AuthCallbackPage() {
   const navigate = useNavigate();
   const [statusMessage, setStatusMessage] = useState('Loading...');
   const [callbackError, setCallbackError] = useState<Error | null>(null);
+  const [isAccountExisting, setAccountAsExisting] = useState(true);
+  const onSubmitRef = useRef(null);
+  const [inFlight, setInFlight] = useState(false);
 
   const [searchParams] = useSearchParams();
 
-  // The signIn should run only once
   useEffect(() => {
     const signInProcess = async () => {
-      if (isSignInWithEmailLink(firebaseAuth, window.location.href)) {
-        const accountId = decodeIfTruthy(searchParams.get('accountId'));
-        const isRecovery = decodeIfTruthy(searchParams.get('isRecovery'));
-        const success_url = decodeIfTruthy(searchParams.get('success_url'));
-        const public_key_lak = decodeIfTruthy(searchParams.get('public_key_lak'));
-        const contract_id = decodeIfTruthy(searchParams.get('contract_id'));
-        const methodNames = decodeIfTruthy(searchParams.get('methodNames'));
+      if (!isSignInWithEmailLink(firebaseAuth, window.location.href)) {
+        navigate('/signup');
```

Comment:
would this navigate inherit all query params?

Reply:
We don't need it because it is going to the signup page of the gateway directly.